packages/frontend/lib/index.ts

```diff
@@ -119,9 +119,9 @@ export default class Nango {
         if (
             options &&
             'credentials' in options &&
-            (!('oauth_client_id_override' in options.credentials) || !('oauth_client_secret_override' in options.credentials)) &&
-            !('token_id' in options.credentials) &&
-            !('token_secret' in options.credentials) &&
+            (('token_id' in options.credentials && 'token_secret' in options.credentials) ||
```

Comment:
As mentioned previously this should be abstracted to be made more readable. Will tackle in a later update.